src/write_gcm.c

```diff
@@ -3869,13 +3869,12 @@ static int wg_transmit_unique_segment(const wg_context_t *ctx,
 
       wg_log_json_message(
           ctx, "Server response (CollectdTimeseriesRequest):\n%s\n", response);
-      // Since the response is expected to be valid JSON, we don't
-      // look at the characters beyond the closing brace.
+      // The response is expected to be valid JSON. In case the response
+      // isn't empty, a potential partial success case may have occured. In
```

Comment:
The original comment was explaining why it was enough to compare 2 characters, so probably best to leave it intact.
As for the rest of the comment, how about:
```c
// When the response isn't empty, it represents a partial success case.
// Because some points were accepted in that case, we treat it as a
// successful API call.
```
? Also below.

Reply:
PTAL.